### PR TITLE
Add blog post  on document databases vs relational databases

### PR DESCRIPTION
**Pull request by Fashander**
# Description

Closes FerretDB/engineering#227.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by ptrfarkas — DISMISSED** [review]

File: website/blog/2024-11-12-relational-vs-document-database-comparison.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  Fields can be added or modified as the application evolves.
+- **Data Encapsulation**: Each document encapsulates data as a single entity, reducing the need for JOIN operations, which are common in relational databases.
+- **Horizontal Scalability**: Document databases are typically built to scale horizontally, making them ideal for distributed systems and high-availability applications.
+- **High Availability**: Built with replication and sharding in mind, document databases are often easier to scale and distribute globally.
+
+## Comparing document databases and relational databases
+
+Here's a quick comparison of the characteristics of document and relational databases:
+
+```markdown
```
Comment (AlekSi) [inline website/blog/2024-11-12-relational-vs-document-database-comparison.md:85]: But that's already a markdown file. Backticks prevent the table from being rendered correctly

**Review by AlekSi — COMMENTED** [review]
Please update the date

File: website/blog/2024-11-12-relational-vs-document-database-comparison.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+[Couchbase](https://www.couchbase.com/) is known for its low-latency access to data, optimized for real-time applications.
+It offers a multi-model approach, combining both document and key-value storage, which suits applications where fast, scalable access to data is essential.
+Couchbase supports SQL-like querying with N1QL, which allows for expressive, flexible queries over JSON documents.
+
+Couchbase's ability to handle data in both document and key-value formats makes it versatile for complex web applications and IoT systems.
+
+### CouchDB
+
+Developed by the Apache Software Foundation,[CouchDB](https://couchdb.apache.org/) is a NoSQL database with an HTTP-based API, offering a RESTful architecture ideal for distributed systems.
```
Comment (AlekSi) [inline website/blog/2024-11-12-relational-vs-document-database-comparison.md:142]:
```suggestion
Developed by the Apache Software Foundation, [CouchDB](https://couchdb.apache.org/) is a NoSQL database with an HTTP-based API, offering a RESTful architecture ideal for distributed systems.
```

File: website/blog/2024-11-12-relational-vs-document-database-comparison.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+### MySQL
+
+[MySQL](https://www.mysql.com/) is a high-performance, open-source relational database, popular for web applications, particularly with the LAMP stack (Linux, Apache, MySQL, PHP).
+MySQL's strength lies in its speed and reliability, handling millions of rows efficiently and supporting numerous storage engines for versatility.
+
+MySQL's broad support and numerous plugins make it a versatile option for a wide range of applications, from content management systems to e-commerce platforms.
+
+### Microsoft SQL Server
+
+A commercial database from Microsoft,[SQL Server](https://www.microsoft.com/en-us/sql-server) integrates well with Windows environments and supports enterprise-grade features such as high availability and replication.
```
Comment (AlekSi) [inline website/blog/2024-11-12-relational-vs-document-database-comparison.md:167]:
```suggestion
A commercial database from Microsoft, [SQL Server](https://www.microsoft.com/en-us/sql-server) integrates well with Windows environments and supports enterprise-grade features such as high availability and replication.
```

File: website/static/img/blog/relational-document.jpg
Comment (AlekSi) [inline website/static/img/blog/relational-document.jpg:1]: Why does this image use ferretdb.IO while #4620 uses ferretdb.COM?

**Review by AlekSi — APPROVED** [review]

**Pull request merged**